### PR TITLE
[#209] Bugfix: GTM only tracks initial page load

### PR DESCRIPTION
**Pull request by RRoussev88**
bugfix issue #209

### CODE REVIEW

File: frontend/src/common/util/useGTM.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { default as TagManager, TagManagerArgs } from 'react-gtm-module'
 
 const gtmId = getConfig().publicRuntimeConfig.GTM_ID
 
-export type EventType = 'event_1' | 'change_language'
+export type EventType = 'event_1' | 'change_language' | 'page_view'
```
Comment (kachar) [inline frontend/src/common/util/useGTM.ts:6]: I think we can delete `event_1` as it's not currently used

File: frontend/src/pages/_app.tsx [hunk 2]
```diff
@@ -13,20 +14,42 @@ import 'styles/global.scss'
...
     // Remove the server-side injected CSS.
     const jssStyles = document.querySelector('#jss-server-side')
     if (jssStyles && jssStyles.parentElement) {
       jssStyles.parentElement.removeChild(jssStyles)
     }
+  }, [])
 
-    // Init GTM
-    useGTM().initialize({
+  useEffect(() => {
```
Comment (kachar) [inline frontend/src/pages/_app.tsx:29]: Good call to use dedicated effect for the gtm :rocket: 

As I see we're going to `initialize` gtm on every language change now.

I can't evaulate if this is okay or not, do you have a view on that?

If it's not okay, we can use a 3rd effect just for the init method with no dependencies

File: frontend/src/pages/_document.tsx [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
 import FaviconMetadata from 'components/brand/FaviconMetadata'
 
+const GTM_ID = getConfig().publicRuntimeConfig.GTM_ID
+
 export default class CustomDocument extends Document {
   render() {
     return (
       <Html lang="en">
         <Head>
+          {/* Google Tag Manager */}
```
Comment (kachar) [inline frontend/src/pages/_document.tsx:16]: I don't get why we have this placed in the document as the module we're using `react-gtm` is doing exactly this job

https://github.com/alinemorelli/react-gtm/blob/b1257405a82e900c174032a773a8ba70a8628081/src/Snippets.js#L12-L22

When we call `initialize` this is what is injected in our `<Head>` dynamically, so we don't have to do it manually.

Comment (RRoussev88) [reply frontend/src/pages/_document.tsx]: Looks like you are right. I should have read the `react-gtm-module` documentation :)

File: frontend/src/pages/_app.tsx (hunk 2 above)
Comment (RRoussev88) [reply frontend/src/pages/_app.tsx]: I decided to put GTM initialization back to the first useEffect after all. It is better to call it only once.

**Review by kachar — APPROVED** [review]
Awesome! :rocket: :100: 

Thanks for the contribution @RRoussev88!

**Pull request merged**